### PR TITLE
[release/3.1] Add CentOS 9 RID and make CentOS 8 arm64 as well

### PR DESCRIPTION
**Pull request by omajid**
# Port Description

This is the only part of https://github.com/dotnet/runtime/pull/34088 that wasn't already backported to release/3.1.

https://www.redhat.com/en/blog/faq-centos-stream-updates says:

> There will not be a CentOS Linux 9.
>
> CentOS Stream 9 will launch in Q2 2021 as part of the RHEL 9 development process.

That's my motivation: I want to be able to build .NET Core 3.1 for RHEL 9

The last part also means that the current parent-child relationship that we use for RHEL->CentOS in our RID graph will flip starting in 9. I dont see any immediate impact, so I think we can leave it alone for now.

I got a (currently internal) build of CentOS Stream 9 and `/etc/os-release` looks like this:

    $ cat /etc/os-release
    NAME="CentOS Stream"
    VERSION="9"
    ID="centos"
    ID_LIKE="rhel fedora"
    VERSION_ID="9"
    PLATFORM_ID="platform:el9"
    PRETTY_NAME="CentOS Stream 9"
    ANSI_COLOR="0;31"
    CPE_NAME="cpe:/o:centos:centos:9"
    HOME_URL="https://centos.org/"
    BUG_REPORT_URL="https://bugzilla.redhat.com/"
    REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux 9"
    REDHAT_SUPPORT_PRODUCT_VERSION="CentOS Stream"

I have been told that builds of CentOS Stream 9 should become public soon. I don't know exactly when, though.

# Customer Impact

This (kind of) blocks Red Hat from building .NET Core 3.1 on RHEL 9: we can apply this patch locally, but it's probably better for everyone if it's here upstream too.

# Regression

This is not a regression.

# Testing

I have tested that this RID update fixes the build issue for me.

# Risk

The risk of this is pretty low: it's just adding RIDs.

### CODE REVIEW

**Comment by danmoseley** [comment]
If this is to be serviced, please add the usual template.
cc @wfurt owns RID updates for us.

**Comment by omajid** [comment]
> If this is to be serviced, please add the usual template.

Okay. I copied the template from another PR in this repo. If it's not right, can you help me find the canonical template?

**Comment by danmoseley** [comment]
Looks great to me. @wfurt are you the right person to evaluate this one?

**Review by wfurt — APPROVED** [review]
LGTM.
It seems like are trying to keep RID graph same across releases and this matches main branch.

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json [hunk 1]
```diff
@@ -200,7 +200,8 @@
         "3.1.3",
         "3.1.4",
         "3.1.5",
-        "3.1.6"
+        "3.1.6",
+        "3.1.7"
       ],
       "BaselineVersion": "3.1.5",
```
Comment (Anipik) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: Can you update the baseline version to 3.1.7 as well

**Review by Anipik — APPROVED** [review]

**Pull request merged**